contributor_docs/README.md

```diff
@@ -140,27 +111,85 @@ A complete guide to unit testing is beyond the scope of the p5.js documentation,
 
-# Building Documentation
 
-Inline comments in p5.js are built into the public-facing [reference manual](https://p5js.org/reference/). You can also view this locally:
+# Gotchas
+
+The developer tooling included with the p5.js codebase is intentionally very strict about some things. This is a good thing! It makes everything consistent, and it will encourage you to be disciplined. This means you may try to change something only to have your commit rejected by the project, but don't get discouraged; even seasoned p5.js developers get caught by this stuff all the time. Typically the problem will be in one of two areas, code syntax or unit tests.
+
+## Code Syntax
+
+p5.js requires clean and stylistically consistent code syntax, which it enforces using tools called [Prettier](https://prettier.io/) and [ESlint](https://eslint.org/). Certain style rules are checked before you commit, but you can also install an [ESlint plugin](https://eslint.org/docs/user-guide/integrations#editors) for your code editor to highlight errors as soon as you type them. In general, we err on the side of flexibility when it comes to code style, in order to lower the barriers to participation and contribution.
+
+To detect errors, run the following command in your terminal (do not type the `$` prompt):
+
+```
+$ npm run lint
+```
+
+Some syntax errors can be automatically fixed:
+
+```
+$ npm run lint:fix
+```
+
+Sticking with the established project style is usually preferable, but [occasionally](https://github.com/processing/p5.js/search?utf8=%E2%9C%93&q=prettier-ignore&type=) using an alternate syntax might make your code easier to understand. For these cases, Prettier [offers an escape hatch](https://prettier.io/docs/en/ignore.html), the `// prettier-ignore` comment, which you can use to make granular exceptions. Try to avoid using this if you can, because there are good reasons for most of the style preferences enforced by the linting.
+
+Here is a quick summary of code style rules. Please note that this list may be incomplete, and it's best to refer to the [.prettierrc](https://github.com/processing/p5.js/blob/master/.prettierrc) and [.eslintrc](https://github.com/processing/p5.js/blob/master/.eslintrc) files for the full list.
+* ES6 code syntax is used
+
+* Use single quotes (rather than double quotes)
+
+* Indentation is done with two spaces
+
+* All variables defined in the code should be used at least once, or removed completely
+
+* Do not compare x == true or x == false. Use (x) or (!x) instead. x == true, is certainly different from if (x)! Compare objects to null, numbers to 0 or strings to "", if there is chance for confusion.
+
+* Comment your code whenever there is ambiguity or complexity in the function you are writing
+
+* See the [Mozilla JS practices](https://developer.mozilla.org/en-US/docs/Mozilla/Developer_guide/Coding_Style#JavaScript_practices) as a useful guide for more styling tips
+
+  
+
+## Unit Tests
```

Comment:
this section is great and I think parts of it could also go in `unit_testing.md`. after this is merged i'll take a look and propose any revisions i think are worthwhile.